src/hibernate-resume/hibernate-resume-generator.c

```diff
@@ -81,6 +98,90 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
+static int parse_efi_hibernate_location(void) {
+        int r = 0;
+
+#if ENABLE_EFI
+        static const JsonDispatch dispatch_table[] = {
+                { "uuid",         JSON_VARIANT_STRING,   json_dispatch_id128,        offsetof(EFIHibernateLocation, id),            JSON_MANDATORY             },
+                { "offset",       JSON_VARIANT_UNSIGNED, json_dispatch_uint64,       offsetof(EFIHibernateLocation, offset),        JSON_MANDATORY             },
+                { "id",           JSON_VARIANT_STRING,   json_dispatch_const_string, offsetof(EFIHibernateLocation, id),            JSON_PERMISSIVE|JSON_DEBUG },
+                { "imageId",      JSON_VARIANT_STRING,   json_dispatch_const_string, offsetof(EFIHibernateLocation, image_id),      JSON_PERMISSIVE|JSON_DEBUG },
+                { "versionId",    JSON_VARIANT_STRING,   json_dispatch_const_string, offsetof(EFIHibernateLocation, version_id),    JSON_PERMISSIVE|JSON_DEBUG },
+                { "imageVersion", JSON_VARIANT_STRING,   json_dispatch_const_string, offsetof(EFIHibernateLocation, image_version), JSON_PERMISSIVE|JSON_DEBUG },
+                {},
+        };
+
+        _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+        _cleanup_free_ char *location_str = NULL, *device = NULL, *id = NULL, *image_id = NULL,
+                       *version_id = NULL, *image_version = NULL;
+        char node[STRLEN("UUID=") + SD_ID128_UUID_STRING_MAX];
+        EFIHibernateLocation location = {};
+        int log_level, log_level_ignore;
+
+        log_level = arg_resume_device ? LOG_DEBUG : LOG_ERR;
+        log_level_ignore = arg_resume_device ? LOG_DEBUG : LOG_WARNING;
```

Comment:
i'd not bother with this, and just always log at LOG_ERR/LOG_WARNING level. If the data is in the efi var i see no shame in reading it, and hence log about it.